kernel/USER_GUIDE.md

```diff
@@ -91,57 +91,85 @@ while(fileIter.hasNext()) {
   FilteredColumnarBatch scanFileColumnarBatch = fileIter.next();
   CloseableIterator<Row> scanFileRows = scanFileColumnarBatch.getRows();
 
-  CloseableIterator<DataReadResult> dataResultIter = Scan.readData(
-    myTableClient,
-    scanStateRow,
-    scanFileRows,
-    Optional.empty() /* filter */
-  );
-
-  while(dataResultIter.hasNext()) {
-    DataReadResult dataReadResult = dataResultIter.next();
-
-    ColumnarBatch dataBatch = dataReadResult.getData();
-
-    // Not all rows in `dataBatch` are in the selected output. An optional selection vector
-    // determines whether a row with a specific row index is in the final output or not.
-    Optional<ColumnVector> selectionVector = dataReadResult.getSelectionVector();
-
-    // access the data for the column at ordinal 0
-    ColumnVector column0 = dataBatch.getColumnVector(0);
-    for (int rowIndex = 0; rowIndex < column0.getSize(); rowIndex++) {
-      // check if the row is selected or not
-      if (selectionVector.isEmpty() || selectionVector.get().getBoolean(rowIndex)) {
-        // Assuming the column type is String.
-        // If it is a different type, call the relevant function on the `ColumnVector`
-        System.out.println(column0.getString(rowIndex));
+  // Get the physical read schema of columns to read from the Parquet data files
+  StructType physicalReadSchema =
+    ScanStateRow.getPhysicalDataReadSchema(tableClient, scanState);
+
+  try (CloseableIterator<Row> scanFileRows = scanFileColumnarBatch.getRows()) {
+    while (scanFileRows.hasNext()) {
+      Row scanFileRow = scanFileRows.next();
+
+      // From the scan file row, extract the file path, size and modification time metadata
+      // needed to read the file.
+      FileStatus fileStatus = InternalScanFileUtils.getAddFileStatus(scanFileRow);
```

Comment:
In a general sense... if we are including this API in an example it seems like it shouldn't be in an internal util package?